static/app/views/dashboards/exportDashboard.tsx

```diff
@@ -0,0 +1,105 @@
+import {addErrorMessage} from 'sentry/actionCreators/indicator';
+
+const delete_properties = ['createdBy', 'dateCreated', 'id', 'dashboardId', 'widgetId'];
+
+async function exportDashboard() {
+  try {
+    const structure = {
+      base_url: null,
+      dashboard_id: null,
+    };
+
+    const params = getAPIParams(structure);
+    const api_url = `https://${params.base_url}/api/0/organizations/testorg-az/dashboards/${params.dashboard_id}/`;
+    const response = await fetch(api_url);
+    const jsonData = await response.json();
+    const normalized = normalize_data(jsonData);
+    normalized.projects = [];
+
+    downloadObjectAsJson(normalized, cleanTitle(normalized.title));
+  } catch (error) {
+    addErrorMessage(
+      'Could not export dashboard. Please wait or try again with a different dashboard'
+    );
+  }
+}
+
+function getAPIParams(structure) {
+  const url = window.location.href;
+  const regex = {
+    base_url: /(\/\/)(.*?)(\/)/,
+    dashboard_id: /(dashboard\/)(.*?)(\/)/,
+  };
+
+  for (const attr in regex) {
+    const match = url.match(regex[attr]);
+    if (match?.length) {
+      structure[attr] = match.length >= 3 ? match[2] : null;
+    }
+  }
+
+  return structure;
+}
+
+function normalize_data(source) {
+  const payload = {};
```

Comment:
So we're getting type errors above because typescript is unsure about the resulting type from this function. I was playing around with it and I think this works. Basically it just says it's going to be a DashboardDetail object but with the deleted keys removed

```js
function normalize_data(
  source: DashboardDetails
): Omit<DashboardDetails, (typeof delete_properties)[number]> {
  const payload: Omit<DashboardDetails, (typeof delete_properties)[number]> = {
    title: '',
    filters: {},
    projects: [],
    widgets: [],
  };
```

`delete_properties` should also be renamed to `deleteProperties` for consistency and we should change the line where it's declared to:

```js
const deleteProperties: readonly string[] = [
  'createdBy',
  'dateCreated',
  'id',
  'dashboardId',
  'widgetId',
] as const;
```

Reply:
Ugh wait, I think there's also a problem with this. Ignore my suggestion for now

Reply:
Kk I updated my comment, the types should work out now

Reply:
This works! Thanks